forum/backends/mysql/models.py

```diff
@@ -0,0 +1,240 @@
+            "body": self.body,
+            "course_id": self.course_id,
+            "comment_thread_id": self.comment_thread_id,
+            "commentable_id": self.commentable_id,
+            "created_at": self.created_at,
+            "updated_at": self.updated_at,
+            "title": getattr(self, "title", None),
+        }
+
+
+class EditHistory(models.Model):
+    """Edit history model class"""
+    DISCUSSION_MODERATION_EDIT_REASON_CODES = [
+    ("grammar-spelling", _("Has grammar / spelling issues")),
+    ("needs-clarity", _("Content needs clarity")),
+    ("academic-integrity", _("Has academic integrity concern")),
+    ("inappropriate-language", _("Has inappropriate language")),
+    ("format-change", _("Formatting changes needed")),
+    ("post-type-change", _("Post type needs change")),
+    ("contains-pii", _("Contains personally identifiable information")),
+    ("violates-guidelines", _("Violates community guidelines")),
+    ]
+    original_body = models.TextField()
+    reason_code = models.CharField(
+        max_length=100,
+        blank=True,
+        null=True,
+        choices=DISCUSSION_MODERATION_EDIT_REASON_CODES
+    )
+    editor_username = models.CharField(max_length=255)
+
+    author = models.ForeignKey(User, related_name="comment_edits", on_delete=models.CASCADE)
+    thread = models.ForeignKey(CommentThread, related_name="comment_threads", on_delete=models.CASCADE, null=True)
+    comment = models.ForeignKey(Comment, related_name="edit_histories", on_delete=models.CASCADE, null=True)
+
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    def to_dict(self):
+        return {
+            "original_body": self.original_body,
+            "reason_code": self.reason_code,
+            "editor_username": self.editor_username,
+            "created_at": self.created_at
+        }
+
+    def __str__(self):
+        return f"Edit by {self.editor_username} on {self.created_at}"
+
+
+class AbuseFlagger(models.Model):
+    """Abuse flagger model class"""
+    content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE)
+    content_object_id = models.PositiveIntegerField()
+    content = GenericForeignKey("content_type", "content_object_id")
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    flagged_at = models.DateTimeField(default=timezone.now)
+
+
+class HistoricalAbuseFlagger(models.Model):
+    """Historical abuse flagger model class"""
+    content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE)
+    content_object_id = models.PositiveIntegerField()
+    content = GenericForeignKey("content_type", "content_object_id")
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    flagged_at = models.DateTimeField(default=timezone.now)
+
+
+def validate_upvote_or_downvote(value):
+    if value not in [1, -1]:
+        raise ValidationError('This field only accepts 1 or -1.')
+
+
+class UserVotes(models.Model):
+    """User votes model class"""
```

Comment:
Add a unique constraint for (user, content)

Reply:
done